src/mlpack/tests/rectangle_tree_test.cpp

```diff
@@ -854,6 +864,244 @@ BOOST_AUTO_TEST_CASE(DiscreteHilbertValueTest)
                                                                   point4), -1);
 }
 
+template<typename TreeType>
+void CheckOverlap(const TreeType& tree)
+{
+  bool success = true;
+
+  // Check if two nodes overlap each other.
+  for (size_t i = 0; i < tree.NumChildren(); i++)
+  {
+    success = true;
+
+    for (size_t j = 0; j < tree.NumChildren(); j++)
+    {
+      if (j == i)
+        continue;
+      success = false;
+      // Two nodes overlap each other if and only if there are no dimension
+      // in which they do not overlap each other.
+      for (size_t k = 0; k < tree.Bound().Dim(); k++)
+      {
+        if ((tree.Child(i).Bound()[k].Lo() >=
+            tree.Child(j).Bound()[k].Hi()) ||
+            (tree.Child(j).Bound()[k].Lo() >=
+            tree.Child(i).Bound()[k].Hi()))
+        {
+          success = true;
+          break;
```

Comment:
There is a function `DoBoundsIntersect` in `tree_test.cpp` that does the same thing, but now that we have two places in the codebase where we are comparing `HRectBound` objects to see that they overlap, do you think we should add a new member to `HRectBound` to encapsulate this functionality?  We could call it `HRectBound::Contains(const HRectBound& other)`, since there is already an overload for a single point instead of a bound.